### PR TITLE
Improv/material UI

### PR DESCRIPTION
**Pull request by iannbing**
## Description

Replace MDC custom UI components with Material-UI components.

Resolves: #413, #557, #554, #540, #598, #380, #613, #311

### Checklist

- [x] I have manually tested the application to make sure the changes don’t cause any downstream issues, which includes making sure `./orchest status --ext` is not reporting failures when Orchest is running.

### CODE REVIEW

**Comment by yannickperrenet** [comment]
Whilst testing the PR I ran into a few issues (I am unsure whether they have all been introduced due to the migration):

1. No items in the left pane menu stay selected (e.g. https://github.com/orchest/orchest/issues/350) after "moving deeper". For example after opening a pipeline the *Pipelines* menu item is no longer selected.
2. Top bar navigator that select projects is broken. Every time I press a left pane menu item it goes back to the project that is ordered first. For example: Create project A, create project B, open a pipeline of project B, click on environments, see that project A is selected.
3. The *LOGS* view of the pipeline editor is broken. It is shifted upwards too much. Similarly, the JupyterLab iframe top is cut off.
4. Seems like changes in https://github.com/orchest/orchest/pull/587 got lost
5. Button to open a pipeline run that was part of a job is gone

Additional issues (which I guess are not due to this PR):

1. Changing the name of a draft job (that was created by duplication) in the jobslist view changes its schedule from cron to now.
2. I get confused when I create a job, but press cancel, see it as "draft" but not having any of my configuration saved.
3. What is the order of the orchest curated examples? They don't seem to be sorted based on stars (which is fine by me, but would be good that the order is explicit in some way).
4. If you go to *settings* > *show onboarding dialog* and you have the quickstart, then closing the window will always to you to the quickstart. Also when you don't have it, then the button "open quickstart" should also import it in my opinion.
5. Stopping the session, whilst in JupyterLab, takes you back to the pipelines view.

**Comment by yannickperrenet** [comment]
To explain:
> Button to open a pipeline run that was part of a job is gone

Screenshot from 2021-12-22 14-18-57

Underneath the "learn more" there used to be a button. Clicking that button would open the pipeline in "read only" mode. This way you can expect the pipeline that has run

**Comment by ricklamers** [comment]
Minor issue: no button spacing on the JupyterLab Configure page

image

**Comment by ricklamers** [comment]
Minor issue: I think the primary action button on the dialog should be solid blue style.

image

**Comment by ricklamers** [comment]
Minor issue: I think the alignment of the icon and text of the button should be vertically centered.

image

**Comment by ricklamers** [comment]
Minor issue: spacing should be equal, no exit logs button (to pipeline editor) in top right.

image

**Comment by ricklamers** [comment]
Minor issue: some spacing between the environment variable listings on the pipeline settings page.

**Comment by ricklamers** [comment]
Minor issue: the toggle seems a bit too small. But feel free to overrule if you disagree here :)

image

**Comment by ricklamers** [comment]
Medium issue: it says it's invalid JSON while it isn't in the job edit view.

image

**Comment by ricklamers** [comment]
Minor issue: the grey color scheme of the search bar makes it feel disabled. This is also the case on the job detail page (when searching among pipeline runs).

image

**Comment by ricklamers** [comment]
Minor issue: I think the logo spacing can be reduced a bit (closer to menu toggle) to make the logo feel less "floaty".

image

**Comment by ricklamers** [comment]
I can only say, _oh my god_. This is so well done, @iannbing you are a hero! You will go down in Frontend History, I'm certain of it. Thank you so much for all the hard work you've put in :pray: :pray: :pray:.

**Comment by ricklamers** [comment]
Minor issue: I think the font size of the project selector is too big. It should hold fairly long project titles fully. Perhaps 70-80% of what it is now?

**Comment by iannbing** [comment]
@ricklamers Thanks for finding so many issues :-)

- [x] Minor issue: no button spacing on the JupyterLab Configure page
- [x] Minor issue: I think the primary action button on the dialog should be solid blue style.
- [x] Minor issue: spacing should be equal, no exit logs button (to pipeline editor) in top right.
- [x] Minor issue: some spacing between the environment variable listings on the pipeline settings page.
- [x] Minor issue: the toggle seems a bit too small. But feel free to overrule if you disagree here :)  (in fact, the combination of Switch + action text is a bit confusing imo, i.e. it took me a sec or two to think "is this text describing a state or an action?", we could experiment further on this use case. )
- [x] Medium issue: it says it's invalid JSON while it isn't in the job edit view.
- [x] Minor issue: the grey color scheme of the search bar makes it feel disabled. This is also the case on the job detail page (when searching among pipeline runs).
- [x] Minor issue: I think the logo spacing can be reduced a bit (closer to menu toggle) to make the logo feel less "floaty".
- [x] Minor issue: I think the font size of the project selector is too big. It should hold fairly long project titles fully. Perhaps 70-80% of what it is now?
- [ ] New issue found: "Running services" menu in PipelineView is not positioned correctly.

Unable to reproduce, requires further confirmation:
- [ ] Minor issue: I think the alignment of the icon and text of the button should be vertically centered.

**Comment by yannickperrenet** [comment]
The logs "display" (both in the *LOGS* views and in the step properties pane) has become significantly smaller, making it less readable and needing to scroll more.

image

**Comment by yannickperrenet** [comment]
When you delete a service then pressing the bucket icon (to delete) will also toggle a open (or close, depending on the previous state) of the row. Instead of you press delete, the row should not open or close.

image

**Comment by yannickperrenet** [comment]
If you go to *settings* > *updates* > *check for updates* then you see:
image

which looks a bit funky and not identical to the styling of other views.

**Comment by yannickperrenet** [comment]
http://localhost:8000/manage-users looks broken:

image

**Comment by yannickperrenet** [comment]
Alignment of *create new variable* button seems of in the *environment variables tab* when creating a job:

image

**Comment by yannickperrenet** [comment]
When you open a file (through the step's properties pane) then you can either open it in JupyterLab or you can press *view file* which opens a html rendered version of the Notebook. However, this is incorrectly embedded in the page:

image

NOTE: There is also no way to close out of this view, which used to be there.

**Comment by yannickperrenet** [comment]
1. Create a job with parameters
2. Add a few parameters
3. Start the job
4. Open the job and inspect its parameters
5. See that the *run specification* says that all runs are *parameterless run* whilst they are not.

image

**Comment by yannickperrenet** [comment]
> > If you go to settings > show onboarding dialog and you have the quickstart, then closing the window will always take you to the quickstart. Also when you don't have it, then the button "open quickstart" should also import it in my opinion.
> 
> The expected behavior goes likes this: if user has quickstart project, the last button "OPEN QUICKSTART PIPELINE". If not, the button will be "GET STARTED", and it simply closes the dialog. Previously the Onboarding dialog has its own state of existing projects, so if you delete quickstart, Onboarding still shows "OPEN QUICKSTART PIPELINE". I connect Onboarding Dialog to ProjectsContext, so now there's no more inconsistency.

This: 
> If you go to settings > show onboarding dialog and you have the quickstart, then closing the window will always take you to the quickstart.

is actually still happening. When I press "close" of the onboarding dialog, then I don't want to be taken to the quickstart pipeline since I explicitly said that is not what I want.

File: services/orchest-webserver/client/src/pipeline-view/PipelineView.tsx [hunk 1]
```diff
@@ -2184,59 +2199,68 @@ const PipelineView: React.FC = () => {
...
-        );
-      }
+      // serviceLink.urls =
 
-      if (urls.length == 0) {
-        serviceLinks.push(
-          <i key={serviceName + "-i"}>This service has no endpoints.</i>
-        );
-      }
+      // for (let url of urls) {
```
Comment (ricklamers) [inline services/orchest-webserver/client/src/pipeline-view/PipelineView.tsx:2245]: Do we want to drop these comments?

Comment (iannbing) [reply services/orchest-webserver/client/src/pipeline-view/PipelineView.tsx]: Removed.

**Pull request merged**